scenarios/cpp/windows/captioning/caption.cpp

```diff
@@ -0,0 +1,580 @@
+			outputStream.open (userConfig.outputFile.value ());
+			outputStream.close ();			
+		}
+	}
+
+	if (false == userConfig.srtEnabled && false == userConfig.partialResultsEnabled)
+	{
+		WriteToConsoleOrFile ("WEBVTT\n\n", userConfig);
+	}
+}
+
+/*
+Main functions
+*/
+
+UserConfig UserConfigFromArgs (int argc, char* argv[])
+{
+	bool prioritizeAccuracyEnabled = false;
+	bool dictationEnabled = false;
+	bool profanityFilterEnabled = false;
+	std::optional<std::vector<std::string>> languageIDLanguages;
+	std::optional<std::string> inputFile;
+	std::optional<std::string> outputFile;
+	std::optional<std::string> phraseList;
+	bool suppressOutputEnabled = false;
+	bool partialResultsEnabled = false;
+	std::optional<std::string> stablePartialResultThreshold;
+	bool srtEnabled = false;
+	bool trueTextEnabled = false;
+	std::string subscriptionKey;
+	std::string region;
+
+// Verify argc >= 3 (caption.exe, subscriptionKey, region)
+	if (argc < 3)
+	{
+		throw TooFewArguments ();
+	}
+
+	subscriptionKey = argv[argc - 2];
+	region = argv[argc - 1];
+
+	prioritizeAccuracyEnabled = CmdOptionExists (argv, argv + argc, "-a");
+	dictationEnabled = CmdOptionExists (argv, argv + argc, "-d");
+	profanityFilterEnabled = CmdOptionExists (argv, argv + argc, "-f");
+	std::optional<std::string> languageIDLanguagesResult = GetCmdOption (argv, argv + argc, "-l");
+	if (true == languageIDLanguagesResult.has_value ())
+	{
+		languageIDLanguages = std::optional<std::vector<std::string>>{ split (languageIDLanguagesResult.value (), ',') };
+	}
+	inputFile = GetCmdOption (argv, argv + argc, "-i");
+	outputFile = GetCmdOption (argv, argv + argc, "-o");
+	phraseList = GetCmdOption (argv, argv + argc, "-p");
+	suppressOutputEnabled = CmdOptionExists (argv, argv + argc, "-q");
+	partialResultsEnabled = CmdOptionExists (argv, argv + argc, "-u");
+	stablePartialResultThreshold = GetCmdOption (argv, argv + argc, "-r");
+	srtEnabled = CmdOptionExists (argv, argv + argc, "-s");
+	trueTextEnabled = CmdOptionExists (argv, argv + argc, "-t");
+	
+	UserConfig userConfig = UserConfig (
+		prioritizeAccuracyEnabled,
+		dictationEnabled,
+		profanityFilterEnabled,
+		languageIDLanguages,
+		inputFile,
+		outputFile,
+		phraseList,
+		suppressOutputEnabled,
+		partialResultsEnabled,
+		stablePartialResultThreshold,
+		srtEnabled,
+		trueTextEnabled,
+		subscriptionKey,
+		region
+	);
+	
+	return userConfig;
+}
+
+std::shared_ptr<Audio::AudioConfig> AudioConfigFromUserConfig (UserConfig userConfig)
+{
+	std::shared_ptr<Audio::AudioConfig> audioConfig;
+	if (true == userConfig.inputFile.has_value ())
+	{
+		audioConfig = Audio::AudioConfig::FromWavFileInput (userConfig.inputFile.value ());
+	}
+	else
+	{
+		audioConfig = Audio::AudioConfig::FromDefaultMicrophoneInput ();
+	}
+	return audioConfig;
+}
+
+std::shared_ptr<SpeechConfig> SpeechConfigFromUserConfig (UserConfig userConfig)
+{
+	std::shared_ptr<SpeechConfig> speechConfig;
+	if (userConfig.languageIDLanguages.has_value ()) {
```

Comment:
Be consistent in putting the open and close curly brackets "{", "}" in a new line